### DISCUSSION TITLE
How to apply masks to TableColumn

### DISCUSSION THREAD

**Discussion by pedrocoelho98** [filamentphp/filament#3047] [Help] [answered]
On forms we can control the TextInput **mask** and define **decimalPlaces**, **decimalSeparator**, **thousandsSeparator**, etc.

Can we control these attributes when displaying the field on the table?

**Comment by danharrin** [comment, marked as answer]
`->formatStateUsing(fn ($state) => number_format($state, 0, '.', ','))`

**Reply by pedrocoelho98** [reply]
Thanks you so much!
I adapted your suggestion to allow for nulls and decimal places without forcing them:
->formatStateUsing(fn ($state) => is_null($state)? '': number_format($state, strlen(substr(strrchr($state, "."), 1)),',', ' '))
